Add a "filtered" event

WyohKnott opened the pull request:
Add a filtered event to apply postprocessing (python/metadata) to posts that have been excluded from download by the filtering rules (image and chapter filters).

The goal is to do analysis on filtered content in order to feed a blacklist.

Commit:
[naver] support video downloads (#7395)

and fix 'date' for posts newer than 24h

- fix naver blogposts directory error for posts created less than 24 hours ago
- add support for videos on naver blogs
- refactor to remove unnecessary imports
- fix video url scraping method
- rework date handling
- rework video extraction
- add 'videos' option
- add test

stephanelsmith committed:
[twitter] added 'followers' extractor

modeled after the 'following' extractor

- cleanup
- add test

Commit:
Add a "filtered" event

Add a filtered event to apply postprocessing (python/metadata) to posts
that have been excluded from download by the filtering rules
(image and chapter filters).

The goal is to do analysis on filtered content in order to feed a blacklist.

Twi-Hard commented:
Does this also work with exec postprocessors? That could be really useful to me.

WyohKnott commented:
> Does this also work with exec postprocessors? That could be really useful to me.

Welll a postprocesssor can be trigerred by such an event, including exec.


Here is my use case:

```json
        "reddit":
        {
            "client-id": "seeeeecret",
            "user-agent": "Python:gallery-dl:v1.0 (by /u/noyoudont)",
            "comments": 0,
            "morecomments": false,
            "date-format": "%Y%m%d",
            "filename": "{date:%Y%m%d}_{author!g}_{subreddit!g}_{title[:120]!g}_{id}.{extension}",
            "directory": ["reddit", "users", "{author}"],
            "recursion": 1,
            "embeds": true,
            "videos": true,
            "parent-metadata": "_reddit",
            "metadata": true,
            "whitelist": ["redgifs", "soundgasm", "ytdl"],
            "image-filter": [
                "not (re.search(r'(' + '|'.join(line.strip() for line in open('gallery-dl/reddit/users/banned_users.txt', 'r')) + r')', author, re.IGNORECASE))",
                "not (re.search(r'(' + '|'.join(line.strip() for line in open('gallery-dl/reddit/users/banned_groups.txt', 'r')) + r')', subreddit, re.IGNORECASE))",
                "not (re.search(r'(M4F|M4A|M4TF|M4TM|M4NB|F4TF|F4TM|F4NB|TF4M|TF4A|TF4F|TM4M|TM4A|TM4F|NB4M|NB4A|NB4F|Script offer)', title, re.IGNORECASE))"
                ],
            "chapter-filter": [
                "not (re.search(r'(' + '|'.join(line.strip() for line in open('gallery-dl/reddit/users/banned_users.txt', 'r')) + r')', author, re.IGNORECASE))",
                "not (re.search(r'(' + '|'.join(line.strip() for line in open('gallery-dl/reddit/users/banned_groups.txt', 'r')) + r')', subreddit, re.IGNORECASE))",
                "not (re.search(r'(M4F|M4A|M4TF|M4TM|M4NB|F4TF|F4TM|F4NB|TF4M|TF4A|TF4F|TM4M|TM4A|TM4F|NB4M|NB4A|NB4F|Script offer)', title, re.IGNORECASE))"
                ],
            "image-unique": true,
            "postprocessors": [
                {
                    "name": "python",
                    "event": "filtered",
                    "function": "gallery-dl/reddit/users/append_and_sort.py:append_sort_deduplicate"
                }
            ]
```

I run append_and_sort.py:append_sort_deduplicate for every user who are caught in the filter. I have a list of banned subreddit in banned_groups.txt, every user who posts on theses subreddits get added to banned_users.txt', with the python script:

```python
def append_sort_deduplicate(metadata):
    string_to_append = metadata["author"] if metadata["author"] != "[deleted]" else None

    # Append the new string to the file
    with open("gallery-dl/reddit/users/banned_users.txt", "a") as file:
        file.write(f"{string_to_append}\n")

    # Read all lines, sort them, remove duplicates, and write them back
    with open("/gallery-dl/reddit/users/banned_users.txt", "r+") as file:
        lines = file.readlines()
        # Strip whitespace and newline, sort, and use set for deduplication
        unique_lines = sorted(set(line.strip() for line in lines))
        
        # Clear the file content
        file.seek(0)
        file.truncate()
        
        # Write unique sorted lines back to the file
        file.writelines(f"{line}\n" for line in unique_lines)
```

Commit:
Add "filtered" event to archive too

Commit:
Limit the "filtered" event to image-filter and chapter-filter...

and not range/unique filters.

Commit:
[boosty] added new direct message extractor

- formatting
- fixed linting formatting errors
- fixed E999 SyntaxError: invalid syntax
- fixed class naming
- fixed mandatory extractor.boosty.metadata as true requirement
- update
  - apply changes
  - add test
  - update docs/supportedsites
- improve 'dialog' pagination logic

mmmpipi committed:
add bilibili User Articles FavList support

- fix whitespace
- fix extractor names
- Add favlist url user check
- apply changes
- add test
- update docs/supportedsites

mikf committed:
replace '\f' in --print arguments with form feed character

to make it easier to use special type format strings on command-line
(#6938)

mikf committed:
improve '\f' format string handling for --print

add a newline only for f-string / \fF format strings,
as it would break any of the others

NecRaul committed:
[b4k] update domain to arch.b4k.dev

mikf committed:
[b4k] keep support for previous 'arch.b4k.co' domain

mikf committed:
merge #6956: [b4k] update domain to 'arch.b4k.dev' (#6955)

mikf committed:
[patreon] support '/profile/creators' URLs

mikf committed:
[pixiv] prevent exceptions during 'comments' extraction (#6965)

- wrap in try-except block
- do not attempt to fetch comments for 'sanity_level' works

mikf committed:
[weebcentral] fix extracting wrong number of chapter pages (#6966)

When downloading multiple chapters at once, all chapters after the first
one would download only as many pages per chapter as the first one had,
due to reusing a cached/shared dict in the wrong way.

mikf committed:
[imhentai] add support (#1660 #3046 #3824 #4338 #5936)

mikf committed:
[vsco] support '/video/' URLs (#4295 #6973)

requires yt-dlp/youtube-dl to handle m3u8 manifests

mikf committed:
[bunkr] provide fallback URLs for 403 download links (#6732 #6972)

mikf committed:
[vsco] improve 'm3u8' handling

mikf committed:
[patreon] extract 'campaign' metadata (#6989)

mikf committed:
[itaku] support gallery section URLs (#6951)

mikf committed:
[weibo] add 'movies' option (#6988)

disable 'movie' downloads by default

mikf committed:
[tests] move 'e621:frontend' tests into regular results/e621.py

having both e621.py and E621.py in the same directory messes with
Windows

https://github.com/mikf/gallery-dl/commit/6e919a36952a91fde78400cc018836be6819ce9c#commitcomment-152557303

mikf committed:
[bunkr] implement fast '--range' support (#6985)

mikf committed:
[imgur] add support for (hidden) personal posts (#6990)

https://imgur.com/user/me
https://imgur.com/user/me/hidden

mikf committed:
[ytdl] support specifying filesystem paths as 'module' (#6991)

mikf committed:
[imgur] replace AuthorizationError exception with logging message

mikf committed:
[dl:http] add 'sleep-429' option (#6996)

mikf committed:
replace 'print()' with 'output.stderr_write("\n")'

mikf committed:
[imhentai] avoid unnecessary HTTP request

no need to fetch a gallery's '/view/' page when the main page contains
all the same data as well

mikf committed:
[archive] implement support for PostgreSQL databases (#6152)

mikf committed:
[docs] add 'psycopg' to optional dependencies

mikf committed:
[archive] add 'archive-table' option (#6152)

mikf committed:
[archive] support PostgreSQL archives for post processors (#6152)

zWolfrost committed:
[discord] add support (#6836)

* first commit

* add --

* skip video embeds

* fix typo

* removed ambiguity

* add category support

* code tweaks

* more reliable embed extraction

* handle 403 errors (testing done)

* added "parent_id" keyword

* added "parent", "parent_type" keywords

the extractor should be now ready to merge!

* removed unnecessary dict unpacking

* added empty text messages extraction

* added "channel_topic"

* even more metadata extraction

can now extract all embeds images & text, as well as server banners. also code is much better.

* added user avatar and banner

* better pagination

* fix regression

* minor tweaks

* Made requested changes

mikf committed:
[hentairox] add support (#7003)

mikf committed:
[archive] fix NameError when SQLite database path doesn't exist

fixes regression introduced in 841bc9f6

mikf committed:
[furry34] add support (#1078 #7018)

mikf committed:
[hentaiera] add support (#3046 #6952 #7020)

mikf committed:
[reddit] restrict subreddit search results (#7025)

mikf committed:
[imhentai] inherit from BaseExtractor

combine all imhentai-like sites into one module

mikf committed:
[imhentai] improve pagination duplicate filtering

mikf committed:
implement 'downloader' options per extractor category

by setting options inside 'http' or 'ytdl' inside extractor options
or inside subcategory options

{
    "extractor": {
        "mastodon": {
            "http": {
                "rate": "10k"
            }
        },
        "mastodon.social": {
            "http": {
                "rate": "100k"
            }
        }
    },
    "downloader": {
        "rate": "100m"
    }
}

Sets download speed to
-  10k for mastodon.social URLs
- 100k for mastodon sites in general
- 100m for all other sites

mikf committed:
[generic] fix config lookups by subcategory

'subcategory' needs to be set before Extractor.__init__() runs
to be included in '_cfgpath'

mikf committed:
[subscribestar] fix 'post' extractor (#6582)

https://github.com/mikf/gallery-dl/issues/6582#issuecomment-2675939669

mikf committed:
[newgrounds] provide 'comment_html' metadata (#7038)

mikf committed:
fix cmdline arguments not overriding extractor-downloader options

mikf committed:
[scripts/pull-request] handle branch already existing

mikf committed:
merge #6768: [boosty] add 'direct-messages' extractor

mikf committed:
merge #6781: [bilibili] add 'user-articles-favorite' extractor (#6725)

mikf committed:
[oauth] improve wording when opening URL

mikf committed:
[furaffinity] use a 1s delay between requests by default (#7054)

CasualYT31 committed:
[tiktok] add support (#3061 #4177 #5646 #6878 #6708)

* Add TikTok photo support

#3061
#4177

* Address linting errors

* Fix more test failures

* Forgot to update category names in tests

* Looking into re issue

* Follow default yt-dlp output template

* Fix format string error on 3.5

* Support downloading videos and audio

Respond to comments
Improve archiving and file naming

* Forgot to update supportedsites.md

* Support user profiles

* Fix indentation

* Prevent matching with more than one TikTok extractor

* Fix TikTok regex

* Support TikTok profile avatars

* Fix supportedsites.md

* TikTok: Ignore no formats error

In my limited experience, this doesn't mean that gallery-dl can't download the photo post (but this could mean that you can't download the audio)

* Fix error reporting message

* TikTok: Support more URL formats

vt.tiktok.com
www.tiktok.com/t/

* TikTok: Only download avatar when extracting user profile

* TikTok: Document profile avatar limitation

* TikTok: Add support for www.tiktokv.com/share links

* Address Share -> Sharepost issue

* TikTok: Export post's creation date in JSON (ISO 8601)

* [tiktok] update

* [tiktok] update 'vmpost' handling

just perform a HEAD request and handle its response

* [tiktok] build URLs from post IDs

instead of reusing unchanged input URLs

* [tiktok] combine 'post' and 'sharepost' extractors

* [tiktok] update default filenames

put 'id' and 'num' first to ensure better file order

* [tiktok] improve ytdl usage

- speed up extraction by passing '"extract_flat": True'
- pass more user options and cookies
- pre-define 'TikTokUser' extractor usage

* [tiktok] Add _COOKIES entry to AUTH_MAP

* [tiktok] Always download user avatars

* [tiktok] Add more documentation to supportedsites.md

* [tiktok] Address review comments

---------

Co-authored-by: Mike Fährmann <mike_faehrmann@web.de>

mikf committed:
[twitter] support 'grok' cards content (#7040)

mikf committed:
[vipergirls] change default 'domain' to 'viper.click' (4166)

https://github.com/mikf/gallery-dl/issues/4166#issuecomment-2684014628

and update general 'domain' handling

mikf committed:
[docs] add 'tiktok' options (#7060)

mikf committed:
[tiktok] add 'audio' option (#7060)

mikf committed:
[tiktok] add 'avatar' option

mikf committed:
[discord] do not match message URLs as channels (#7059)

mikf committed:
[tiktok] fix variable name

fixes regression introduced in 5e87aee32dde13dc388ca76c3d84d7af1d878e3f

zWolfrost committed:
[discord] add single message support

okbuddyagi committed:
[bunkr] bypassed URL encryption

- fixed imports
- formatting
- revert unrelated formatting changes
- cleanup / simplify

mikf committed:
[erome] extract 'tags' metadata (#7076)

mikf committed:
[sankaku] fix extraction (#7071 #7072)

omit 'Platform: web-app' API header to get sankaku to include
'file_url' data in API responses again

mikf committed:
[chevereto] fix extraction (#7073 #7079)

extract file URLs from an alternate location
if 'og:image' is not set

mikf committed:
[docs] update supportedsites

mikf committed:
merge #7067: [discord] add single message support (#7059)

mikf committed:
merge #7070: [bunkr] decrypt file URLs (#7058 #7085)

mikf committed:
[dl:ytdl] log ytdl module and version when importing

mikf committed:
[archive] fix exception when specifying path as list (#7088)

mikf committed:
release version 1.29.0

mikf committed:
[erome] fix AttributeError for albums without tags (#7076)

fixes regression introduced in 07a9dca459e1a589aa5c81881307fd2e0da9465f

mikf committed:
[text] allow calling 'extract_iter' with invalid arguments

mikf committed:
[docs] fix 'tiktok-range' default value (#7098)

mikf committed:
[tiktok] improve 'tiktok-range' parsing (#7098)

replace values evaluating as False (except 0) with ""

mikf committed:
[jpgfish] decrypt file URLs (#7073 #7079)

mikf committed:
[danbooru][e621] remove constructors

mikf committed:
[danbooru:pool] download posts in pool order (#7091)

- add 'order-posts' option
- add 'num' metadata field for pool position
- update default filenames to order by pool position

mikf committed:
[dl] warn about invalid 'subcategory' values (#7103)

prevent fatal exception when collecting downloader options

mikf committed:
[bunkr] update endpoint (#7097)

mikf committed:
[tenor] add initial support (#6075)

mikf committed:
[vsco] fix 'video' extractor (#7113)

fixes regression introduced in 6420210b0faed439ccb102d195096c48918fe84d

mikf committed:
[vsco] fix extracting videos from '/gallery' results (#7113)

mikf committed:
[dl] improve invalid 'subcategory' value warning (#7103)

mikf committed:
[util] implement 'decrypt_xor()'

mikf committed:
[furaffinity] fix 'artist' metadata (#7115 #6582)

https://github.com/mikf/gallery-dl/issues/6582#issuecomment-2696619061

mikf committed:
[tenor] add 'user' extractor (#6075)

mikf committed:
[workflows:tests] run on ubuntu-22.04

> Ubuntu 20.04 LTS runner will be removed on 2025-04-01

mikf committed:
[furaffinity] improve 'artist_url' extraction (#7115 #7123)

mikf committed:
[util] move Cloudflare/DDoS-Guard detection into 'detect_challenge()'

mikf committed:
[dl:http] detect Cloudflare/DDoS-Guard challenges (#7066 #7121)

mikf committed:
[erome:search] recognize all input URL query parameters (#7125)

mikf committed:
[tenor] support URLs with language codes (#6075)

mikf committed:
[tenor] support '/official/' user URLs (#6075)

mikf committed:
[redgifs:search] support '/search?query=...' URLs (#7118)

mikf committed:
[sankaku] fix search tag limit check

mikf committed:
[sankaku] increase wait time on 429 errors (#7129)

to 10 minutes

mikf committed:
[bunkr] add 'endpoint' option (#7097)

mikf committed:
[reddit] add 'selftext' option (#7111)

mikf committed:
[tenor] rename 'content_description' to just 'description'

mikf committed:
[tenor] relax '/view/' URL pattern (#6075)

mikf committed:
release version 1.29.1

mikf committed:
[twitter] prevent exception in '_extract_components()' (#7139)

mikf committed:
[text] add second argument to 'parse_query_list()' (#7138)

return only values whose name is in 'as_list' as a list

mikf committed:
[facebook] improve 'date' extraction (#7151)

use 'created_time' as alternative when 'publish_time' isn't available

mikf committed:
[civitai] fix/improve query parameter handling (#7138)

mikf committed:
[sankaku] update API URLs (#7154 #7155)

and fix errors due to other changes

mikf committed:
[sankaku] fix categorized tags for posts with >100 tags (#7155)

Deer-Spangle committed:
[furaffinity] Adding a FuraffinityFolderExtractor, which extracts a single folder

- Ensure FuraffinityGalleryExtractor doesn't detect folder links
- Fix example URL for folder extractor
- Reordering classes a bit
- Another tweak of the regex
- One more go at the regex..
- cleanup

mikf committed:
merge #7159: [furaffinity] add 'folder' extractor (#1817)

mikf committed:
[furaffinity] extract 'scraps' metadata (#7015)

boolean value indicating whether a post is part of a user's Scraps
folder or the main gallery

mikf committed:
[tests] add support for skipping an extractor result test

mikf committed:
[sankaku] fix 'tags' metadata (#7155)

rename 'tag_names' to 'tags'

mikf committed:
[tiktok] implement audio extraction without ytdl

mikf committed:
[wikimedia] add 'subcategories' option (#2340)

https://github.com/mikf/gallery-dl/pull/2340#issuecomment-2707177295

mikf committed:
[sankaku] fix potential infinite loop (#7155)

https://github.com/mikf/gallery-dl/issues/7155#issuecomment-2723019761

mikf committed:
[batoto] add 'domain' option (#7174)

allow legacy domains by default

hdk5 committed:
[arcalive] add support (#5657 #7100)

* [arca.live] Add extractor skeleton

* [arcalive] update names and formatting

* [arcalive] implement initial file extraction code

* [arcalive] improve '_extract_media()' performance

compile and cache regex on demand

* [arcalive] improve image extraction

- extract 'data-originalurl' URLs if available
- replace URL query strings with 'type=orig'
- ignore emoticons by default

* [arcalive] update defaults

- include 'title' in filenames
- use 0.5-1.5s delay between requests

* [arcalive] use ext from 'data-orig' if available

* [arcalive] update docs/supportedsites

* [arcalive] add tests

* [arcalive] update 'board' extractor pattern

so it doesn't also match 'post' URLs

---------

Co-authored-by: Mike Fährmann <mike_faehrmann@web.de>

mikf committed:
[arcalive] add 'user' extractor (#5657)

mikf committed:
[tests] improve error message of multi type/value tests

improvement of 42070240ae509597b3f3ef6e3cba0325aa61c44c

mikf committed:
[arcalife] add 'gifs' option (#5657)

mikf committed:
release version 1.29.2

CasualYT31 committed:
[tiktok] Add retry mechanism to rehydration data extraction

#7098

mikf committed:
[config] fix using same key multiple times with 'apply' (#7127)

mikf committed:
[deviantart:stash] fix legacy sta.sh links (#7181)

follow redirect instead of rewriting them to deviantart.com/stash/…

mikf committed:
[deviantart] match '/gallery/recommended-for-you' URLs (#7168)

CasualYT31 committed:
[tiktok] Incorporate --retries and --http-timeout into rehydration data extraction

mikf committed:
[zerochan] parse JSON-LD data (#7178)

mikf committed:
[common] add 'subdomains' argument to 'cookies_check()' (#7188)

mikf committed:
[patreon] include subdomains in 'session_id' cookie check (#7188)

mikf committed:
[tests] implement expected failures

mikf committed:
[patreon] do not match '/messages' URLs as creators (#7187)

CasualYT31 committed:
[tiktok] Queue links from user profiles instead of returning a list of links

[tiktok] Move avatar download to the user extractor, which results in more accurate metadata output (it would previously write the metadata of the video which the avatar was scraped from)
[tiktok] Fix tests and remove redundant user profile test

CasualYT31 committed:
[tiktok] Address review comments on _extract_rehydration_data()

CasualYT31 committed:
[tiktok] Don't skip last retry

mikf committed:
[kemonoparty] enable 'username'/'user_profile' metadata by default

BishopRed90 committed:
Adding functionality for subfolder support from Deviant Art

BishopRed90 committed:
Merge pull request #2 from BishopRed90/DA_Subfolder_Support

[DeviantArt] Adding support for sub-folders

mikf committed:
[subscribestar] extract 'title' metadata (#7219)

mikf committed:
[subscribestar] detect redirects to '/age_confirmation_warning' pages

mikf committed:
[aes] simplify 'block_count' calculation

https://github.com/yt-dlp/yt-dlp/pull/12526/commits/abfb3e2277920144667ed4e6aef25c6a7fd6386a

mikf committed:
[kemonoparty] extract 'archives' metadata (#7195)

add 'archives' option for additional data

mikf committed:
[kemonoparty] use '"type": "archive"' for 'archives' files (#7195)

mikf committed:
[bbc] extract more metadata (#6582)

https://github.com/mikf/gallery-dl/issues/6582#issuecomment-2745905375

- title
- title_image
- description
- synopsis

mikf committed:
[deviantart] update code formatting

mikf committed:
[deviantart] match subfolders by name

mikf committed:
[deviantart] add 'subfolders' option

mikf committed:
[deviantart] add 'subfolder' tests

Commit:
[mangapark] utilizing more graphql and adjust functionality for new site (#4999)

- undo formatting changes
- simplify code
- update and fix tests

mikf committed:
[hitomi] fix extractors (#7230)

mikf committed:
[bunkr] fix filename extraction (#7237)

mikf committed:
[imhentai] support 'hentaienvy.com' and 'hentaizap.com' (#7192 #7218)

and move 'hentaifox' support to this module as well

mikf committed:
merge #7220: [deviantart] add subfolder support (#4988 #7185)

mikf committed:
merge #7232: [mangapark] fix extractors (#4999 #5883 #6507 #6908)

DjofPlusPlus committed:
Handle instagram.com/share/ URLs

- added Sec-Fetch* headers for proper redirect
- fixed formatting
- restore 'guide' test
- integrate in PostExtractor code

mikf committed:
[nozomi] fix extractors (#7242)

update internal domains to 'gold-usergeneratedcontent.net'

mikf committed:
[mangapark] support mirror domains

mikf committed:
[mangapark] support v3 URLs (#2072)

mikf committed:
[kemonoparty:discord] look up 'channel_name' when missing (#7245)

mikf committed:
[skeb] make exceptions when extracting posts non-fatal (#7250)

mikf committed:
[danbooru] add 'favgroup' extractor

mikf committed:
[arcalive] extend 'gifs' option

use fallback URLs by default

mikf committed:
[dl:ytdl] support processing inline HLS/DASH manifests (#6379 #7006)

mikf committed:
[instagram] download videos from 'video_dash_manifest' data

(#6379 #7006)

requires yt-dlp / youtube-dl

mikf committed:
[mastodon] support '/statuses' URLs (#7255)

- /statuses/123456789
- /users/USER/statuses/123456789

mikf committed:
[sexcom] remove constructors

mikf committed:
[sexcom] fix 'gif' pin extraction (#7239)

with much less metadata

mikf committed:
[sexcom] support new-style '/gifs' URLs (#7239)

mikf committed:
[sexcom] support new-style '/videos' URLs (#7239)

mikf committed:
[util] update auto Firefox version

BishopRed90 committed:
Adding support for the Multi Image Gallery

BishopRed90 committed:
Adding support for the Multi Image Gallery

BishopRed90 committed:
Removing a test flag (is_downloadable) to true.

BishopRed90 committed:
Fixing formatting of a line that is too long

mikf committed:
[sexcom] extract 'date_url' metadata (#7239)

mikf committed:
merge #7191: [tiktok] add retry mechanism to rehydration data extraction

mikf committed:
[deviantart] update

- use 'index_file' for additional file IDs
- rename 'img_count' to 'count'
- update 'archive_fmt'

mikf committed:
[deviantart] add multi-file test

mikf committed:
merge #7261: [deviantart] support multiple images for single posts

(#6653)

mikf committed:
[pinterest] ignore 'story_pin_static_sticker_block' blocks (#7251)

mikf committed:
release version 1.29.3

mikf committed:
[E621:pool] fix AttributeError (#7265)

fixes regression introduced in 24bbcbcfa3fa5e7a18b431611dbce9f3f70385aa

mikf committed:
[readcomiconline] fix 'issue' extractor (#7269)

- pth -> pht
- spaces -> tabs

mikf committed:
[everia] improve/fix image extraction (#7270)

mikf committed:
[deviantart] fix "KeyError: 'has_subfolders'" (#7272)

mikf committed:
[instagram] use Chrome User-Agent by default (#6379)

mikf committed:
[webtoons] update & simplify code

mikf committed:
[zerochan] fix "KeyError: 'author'" (#7282)

fixes regression introduced in d746e025a0823b5bb04e645bb58c7d1abce013ae

mikf committed:
[webtoons] add 'artist' extractor (#7274)

mikf committed:
[chevereto] support 'imagepond.net' (#7278)

mikf committed:
[dl:ytdl] fix exception when ytdl reports an error (#7301)

when using the HLS/DASH download functionality

mikf committed:
[kemonoparty] fix file paths with backslashes (#7321)

mikf committed:
[issuu] fix extractors (#7317)

astringa1234 committed:
[zzup] support 'w.zzup.com' URLs

- zzup.py support for w.zzup
- fix pattern
- update tests

mikf committed:
[pp:metadata] fix handling of empty directory paths (#7296)

mikf committed:
[rule34xyz] update to API v2 (#7289)

mikf committed:
merge #7327: [zzup] support 'w.zzup.com' URLs

mikf committed:
[pp:metadata] add 'metadata-path' option (#6582)

https://github.com/mikf/gallery-dl/issues/6582#issuecomment-2781643835

mikf committed:
[webtoons] use a default delay pf 0.5-1.5s between requests (#7329)

mikf committed:
[webtoons] download JPEG files in better quality

add 'quality' option

mikf committed:
[formatter] add 'i' and 'f' conversions (#6582)

https://github.com/mikf/gallery-dl/issues/6582#issuecomment-2792166608

mikf committed:
[pp:ugoira] preserve 'extension' when using "mode": "archive" (#7304)

mikf committed:
[tests] add reddit post with redgifs embed

mikf committed:
[gelbooru] fix video URLs (#7345)

update subdomain to 'img4'

w4tchdoge committed:
[pixiv] match phixiv.net URLs

zWolfrost committed:
[discord] fix categories name bug

mikf committed:
[pixiv] update 'work' extractor pattern & add tests

mikf committed:
merge #7352: [pixiv] support 'phixiv.net' URLs

mikf committed:
merge #7353: [discord] fix "parent" keyword inconsistency (#7341)

mikf committed:
[hentai2read] fix exception for chapters without artist (#7355)

mikf committed:
[tumblr] support URLs without subdomain (#7358)

mikf committed:
release version 1.29.4

mikf committed:
use internal, non-caching version of re.compile for extractor patterns

speeds up total compile time of extractor patterns by ~10ms

mikf committed:
[util] update auto Chrome version

mikf committed:
[bluesky:likes] use 'repo.listRecords' endpoint (#7194 #7287)

requires one additional API request per post,
but allows accessing likes of all users without login
and avoids the infinite loop at the end of 'getActorLikes'

add 'endpoint' option

mikf committed:
[bluesky] use self.groups / remove constructors

mikf committed:
[bluesky:likes] fix infinite loop (#7194 #7287)

at the end of 'feed.getActorLikes'

mikf committed:
[bluesky] add 'video' extractor (#4438)

rrockey committed:
[everia] fix 'title' extraction

Due to a recent site change, post titles where no longer being
downloaded into their own directory.

After reviewing the site code changing  extractor to use the "</h2>"
tag instead of the original </h1> resolves this issue.

-----

Generalizing to "</h" works as well and may account for future changes.

mikf committed:
merge #7379: [everia] fix 'title' extraction

mikf committed:
[instagram] add 'followers' extractor (#7374)

mikf committed:
[pp:ugoira] use '_ugoira_frame_index' to detect '.zip' files

mikf committed:
[pp:ugoira] restore 'keep-files' functionality (#7304)

fixes regression introduced in a08c35ad7990a3c532fbe327cd30ace3b416eaa7

mikf committed:
[pp:ugoira] support '"keep-files": true' + custom extension (#7304)

for '"mode": "archive"'

mikf committed:
[fapello] stop pagination on empty results (#7385)

mikf committed:
[moebooru:pool] replace underscores in pool names (#4646)

https://github.com/mikf/gallery-dl/issues/4646#issuecomment-2815698517

mikf committed:
[common] add 'request_location()' convenience function

mikf committed:
[twitter] update API endpoint query hashes (#7382 #7386)

and associated 'variables', 'features', and 'fieldToggles' parameters

mikf committed:
[fanbox] fix 'comments' extraction

mikf committed:
merge #7241: [instagram] handle '/share/' URLs

generated by the mobile app

mikf committed:
merge #6331: [twitter] add 'followers' extractor

mikf committed:
[gelbooru] don't hardcode image server domains (#7392)

mikf committed:
[deviantart] simplify 'additionalMedia' extraction

mikf committed:
[deviantart] fix multi-image URLs (#6653)

https://github.com/mikf/gallery-dl/issues/6653#issuecomment-2816560271

parasiteoflife committed:
fix: Wikifeet - support redesign

mikf committed:
[wikifeet] fix 'bdate' parsing

and update 'shoesize' test results

mikf committed:
merge #7396: [wikifeet] support redesign

mikf committed:
merge #4682: [naver] support videos & fix recent 'date' bug (#7395)

mikf committed:
[deviantart:stash] extract more metadata (#7397)

- stash_id
- stash_name
- stash_description
- stash_folder
- stash_parent

mikf committed:
[kemonoparty] use '/posts-legacy' endpoint (#6780 #6931 #7404)

instead of the "official", broken, and inconsistent endpoint
for creator posts

mikf committed:
[scrolller] retrieve data of single posts via GraphQL

nuhonda committed:
Adding in a first pass at a pictoa extractor

Adds support for galleries and individual Images

Deer-Spangle committed:
[itaku] Add a new extractor for Itaku stars

Deer-Spangle committed:
[weasyl] Set gallery-dl user agent to match danbooru user agent, when making weasyl requests.

To fit with their bot ettiquette: https://projects.weasyl.com/bot-etiquette

mikf committed:
[scrolller] support album posts (#7339)

mikf committed:
[pictoa] update

- simplify code
- update URL patterns
- update tests
- update docs/supportedsites

mikf committed:
merge #7409: [pictoa] add support (#6683)

mikf committed:
merge #7412: [weasyl] use 'gallery-dl' User-Agent header

mikf committed:
[itaku:stars] update

- remove type hints
- add test
- update docs/supportedsites

mikf committed:
merge #7411: [itaku] add 'stars' extractor
